File: javascript/7kyu/Bubblesort Once.js
// https://www.codewars.com/kata/56b97b776ffcea598a0006f2

function bubblesortOnce(arr) {
    let newArr = arr.slice();
    for (let i = 0; i < newArr.length - 1; i++) {
      if (newArr[i] > newArr[i + 1]) {
        [newArr[i], newArr[i + 1]] = [newArr[i + 1], newArr[i]];
      }
    }
    return newArr;
  }

File: java/7kyu/Bubblesort Once.java
// https://www.codewars.com/kata/56b97b776ffcea598a0006f2

public class BubblesortOnce {
  public static int[] bubbleSortOnce(int[] array) {
    int[] result = array.clone();
    for (int i = 0; i < result.length - 1; i++) {
      if (result[i] > result[i + 1]) {
        int temp = result[i];
        result[i] = result[i + 1];
        result[i + 1] = temp;
      }
    }
    return result;
  }
}